src/WebJobs.Script.WebHost/Metrics/LinuxContainerMetricsPublisher.cs

```diff
@@ -251,8 +251,8 @@ public void Initialize()
         public void Start()
         {
             Initialize();
-            _processMonitorTimer = new Timer(OnProcessMonitorTimer, null, TimeSpan.Zero, _memorySnapshotInterval);
-            _metricsPublisherTimer = new Timer(OnFunctionMetricsPublishTimer, null, TimeSpan.Zero, _metricPublishInterval);
+            _processMonitorTimer = new Timer(OnProcessMonitorTimer, null, TimeSpan.FromSeconds(2), _memorySnapshotInterval);
```

Comment:
Hmmm, I wonder why the other ProcessMonitor we start immediately isn't showing up in your analysis (code [here](https://github.com/Azure/azure-functions-host/blob/826c0b520c1e81d2f3106d08711c08f1bce4aa95/src/WebJobs.Script/Scale/HostPerformanceManager.cs#L48)). This guy also starts immediately and starts taking CPU samples. There's another one started for the OOP worker.

Reply:
Also, what will the effect be of us having no memory snapshots taken before functions start executing? We don't know the memory usage for the first 2 seconds of invocations.

Reply:
Looks like that process monitor starts before specialization unlike the metrics one which happens in the middle of specialization. So it probably doesnt show up prominently in the cold start profile.

Yes, this can lead to some imprecision in the metering but it should self correct quickly so the impact should be minimal. Checked with Divya also on this to make sure we are okay doing this.